internal/resources/policy/type/custom/input_schema.go

```diff
@@ -0,0 +1,221 @@
+			}
+		}
+	}
+
+	if v, ok := inputData[reciperesource.TMCExternalIPSKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			inputRecipeData = &inputRecipe{
+				recipe:              tmcExternalIPSRecipe,
+				inputTMCExternalIps: reciperesource.ConstructTMCExternalIPS(v1),
+			}
+		}
+	}
+
+	if v, ok := inputData[reciperesource.TMCHTTPSIngressKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			inputRecipeData = &inputRecipe{
+				recipe:               tmcHTTPSIngressRecipe,
+				inputTMCHTTPSIngress: reciperesource.ConstructTMCHTTPSIngress(v1),
+			}
+		}
+	}
+
+	if v, ok := inputData[reciperesource.TMCRequireLabelsKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			inputRecipeData = &inputRecipe{
+				recipe:                tmcRequireLabelsRecipe,
+				inputTMCRequireLabels: reciperesource.ConstructTMCRequireLabels(v1),
+			}
+		}
+	}
+
+	return inputRecipeData
+}
+
+func flattenInput(inputRecipeData *inputRecipe) (data []interface{}) {
+	if inputRecipeData == nil {
+		return data
+	}
+
+	flattenInputData := make(map[string]interface{})
+
+	switch inputRecipeData.recipe {
+	case tmcBlockNodeportServiceRecipe:
+		flattenInputData[reciperesource.TMCBlockNodeportServiceKey] = reciperesource.FlattenTMCBlockNodeportService(inputRecipeData.inputTMCBlockNodeportService)
+	case tmcBlockResourcesRecipe:
+		flattenInputData[reciperesource.TMCBlockResourcesKey] = reciperesource.FlattenTMCBlockResources(inputRecipeData.inputTMCBlockResources)
+	case tmcBlockRolebindingSubjectsRecipe:
+		flattenInputData[reciperesource.TMCBlockRolebindingSubjectsKey] = reciperesource.FlattenTMCBlockRolebindingSubjects(inputRecipeData.inputTMCBlockRolebindingSubjects)
+	case tmcExternalIPSRecipe:
+		flattenInputData[reciperesource.TMCExternalIPSKey] = reciperesource.FlattenTMCExternalIPS(inputRecipeData.inputTMCExternalIps)
+	case tmcHTTPSIngressRecipe:
+		flattenInputData[reciperesource.TMCHTTPSIngressKey] = reciperesource.FlattenTMCHTTPSIngress(inputRecipeData.inputTMCHTTPSIngress)
+	case tmcRequireLabelsRecipe:
+		flattenInputData[reciperesource.TMCRequireLabelsKey] = reciperesource.FlattenTMCRequireLabels(inputRecipeData.inputTMCRequireLabels)
+
+	case unknownRecipe:
+		fmt.Printf("[ERROR]: No valid input recipe block found: minimum one valid input recipe block is required among: %v. Please check the schema.", strings.Join(recipesAllowed[:], `, `))
+	}
+
+	return []interface{}{flattenInputData}
+}
+
+func validateInput(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
+	value, ok := diff.GetOk(specKey)
+	if !ok {
+		return fmt.Errorf("spec: %v is not valid: minimum one valid spec block is required", value)
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return fmt.Errorf("spec data: %v is not valid: minimum one valid spec block is required among: %v", data, strings.Join(recipesAllowed[:], `, `))
+	}
```

Comment:
Can we define all these error formats at the policy level and reuse them by passing down the variables to them?